governance/program/src/tools/bpf_loader_upgradeable.rs

```diff
@@ -52,3 +56,43 @@ pub fn set_program_upgrade_authority<'a>(
         ],
     )
 }
+
+/// Asserts the program  is upgradable and its upgrade authority is a signer of the transaction
+pub fn assert_program_upgrade_authority_is_signer(
+    program_address: &Pubkey,
+    program_data_info: &AccountInfo,
+    program_upgrade_authority_info: &AccountInfo,
+) -> Result<(), ProgramError> {
+    if program_data_info.owner != &bpf_loader_upgradeable::id() {
+        return Err(ProgramError::IncorrectProgramId);
+    }
+    let program_data_address = get_program_data_address(program_address);
+
+    if program_data_address != *program_data_info.key {
+        return Err(GovernanceError::InvalidProgramDataAccountAddress.into());
+    }
+
+    let upgrade_authority = match deserialize(&program_data_info.data.borrow())
```

Comment:
nit: since only one condition is matched, you can use `let upgrade_authority = if let UpgradeableLoaderState::ProgramData {} = deserialize() { } else { None }`

Reply:
Done in d2c761d0